DIGNDB.App.SmitteStop.API/Attributes/UploadKeysAuthorizationAttribute.cs

```diff
@@ -0,0 +1,82 @@
+﻿using DIGNDB.App.SmitteStop.Core.Contracts;
+using DIGNDB.App.SmitteStop.Domain.Configuration;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using System;
+using DIGNDB.App.SmitteStop.API.Contracts;
+using AnonymousTokens.Server.Protocol;
+using AnonymousTokens.Core.Services.InMemory;
+using Org.BouncyCastle.Crypto.EC;
+using Org.BouncyCastle.Asn1.X9;
+using System.Security.Cryptography;
+
+namespace DIGNDB.App.SmitteStop.API.Attributes
+{
+    public class UploadKeysAuthorizationAttribute : ActionFilterAttribute
+    {
+        private readonly AppSettingsConfig _appSettingsConfig;
+        private readonly AuthOptions _authOptions;
+        private readonly IJwtValidationService _jwtValidationService;
+        private readonly ITokenVerifier _tokenVerifier = new TokenVerifier(new InMemorySeedStore());
+        private readonly InMemoryPrivateKeyStore _privateKeyStore = new InMemoryPrivateKeyStore();
+
+        public UploadKeysAuthorizationAttribute(AppSettingsConfig appSettingsConfig, AuthOptions authOptions, IJwtValidationService jwtValidationService)
+        {
+            _appSettingsConfig = appSettingsConfig;
+            _authOptions = authOptions;
+            _jwtValidationService = jwtValidationService;
+        }
+
+        public override async void OnActionExecuting(ActionExecutingContext context)
+        {
+            if (!_authOptions.AuthHeaderCheckEnabled) return;
+
+            var logger = context.HttpContext.RequestServices.GetService<ILogger<DiagnosticKeysController>>();
+            try
+            {
+                string authHeader = context.HttpContext.Request.Headers["Authorization"];
+                if (authHeader != null && authHeader.Contains("Bearer"))
+                {
+                    var token = authHeader.Replace("Bearer", string.Empty);
+                    token = token.Trim();
+
+                    _jwtValidationService.IsTokenValid(token);
+                    return;
+
+                }
+                else if (authHeader != null && authHeader.StartsWith("Anonymous ")) 
+                {
+                    var _ecParameters = CustomNamedCurves.GetByOid(X9ObjectIdentifiers.Prime256v1);
+                    string[] anonymousToken = authHeader.Replace("Anonymous ", string.Empty).Split(".");
+
+                    var W = _ecParameters.Curve.DecodePoint(Convert.FromBase64String(anonymousToken[0]));
```

Comment:
Why not give this `W` more meaningful name e.g. `elipticCurvePoint`?

Reply:
I will still argue about it.

You are using `BouncyCastle` library and you're calling `DecodePoint` method from it.  
Have you looked at the source code of it or the documentation comments of this method?
Documentation comment of this method is as follows:
```
/**
* Decode a point on this curve from its ASN.1 encoding. The different
* encodings are taken account of, including point compression for
* <code>F<sub>p</sub></code> (X9.62 s 4.2.1 pg 17).
* @return The decoded point.
*/
```
As you can see, it returns `The decoded point`, so why lose this information and rename it to `W`? 
Inside this `DecodePoint` method, local variable which is being returned is called `p` and there's no trace of `W` there.

Reply:
`W` could indeed be named otherwise in the application for more readability. I would suggest `token`.